File: agent_services/python/src/jiji/services/health_check_service.py
import health_check_pb2
import health_check_pb2_grpc

from jiji.services.abstract_service import AbstractService

class HealthCheckService(AbstractService, health_check_pb2_grpc.HealthCheckServiceServicer):

    def Status(self, request, context):
        return health_check_pb2.GetStatusResponse(status="OK")

File: src/jiji/rpc/services/health_check_service.rb
# coding: utf-8

require 'grpc'
require 'health_check_pb'
require 'health_check_services_pb'
require 'jiji/rpc/services/rpc_service_mixin'

module Jiji::Rpc::Services
  class HealthCheckService < Jiji::Rpc::HealthCheckService::Service

    include Encase
    include RpcServiceMixin

    def status(request, call)
      Jiji::Rpc::GetStatusResponse.new(status: 'OK')
    rescue Exception => e # rubocop:disable Lint/RescueException
      handle_exception(e, call)
      return Google::Protobuf::Empty.new
    end

  end
end
